lib/addons.js

```diff
@@ -7,46 +9,50 @@ function Addons (heroku, postgres) {
+      addon = yield this.heroku.apps(app.name).addons().create({attachment: {name: attachment.name}, plan: addon.plan.name});
       console.log('done');
       return addon;
     } catch (err) {
-      console.error(`Skipped addon ${addon.plan.name}`);
-      if (err.body) {
-        console.error(` !    ${err.body.message}`);
-      } else {
-        console.error(err);
-      }
+      console.error(`Skipped addon ${cli.color.magenta(addon.plan.name)}`);
+      if (err.body) cli.error(err.body.message);
+      else          cli.error(err);
     }
   },
 
-  getAddons: function* (app) {
-    let addons = yield this.heroku.request({path: `/apps/${app.name}/addons`});
-    return _.partition(addons, function (addon) {
-      return addon.plan.name.startsWith('heroku-postgresql');
-    });
+  attachAddon: function* (app, addon, attachment) {
+    try {
+      process.stdout.write(`Attaching addon ${cli.color.magenta(attachment.name)} from ${cli.color.cyan(addon.app.name)} to ${cli.color.cyan(app.name)} as ${cli.color.green(attachment.name)}... `);
+      attachment = yield this.heroku.addonAttachments().create({addon: addon.id, app: app.id, name: attachment.name});
+      console.log('done');
+      return attachment;
+    } catch (err) {
+      console.error(`Skipped attachment ${cli.color.magenta(attachment.name)}`);
+      if (err.body) cli.error(err.body.message);
+      else          cli.error(err);
+    }
   },
 
   copyAddons: function* (oldApp, newApp, skipPG) {
-    let allAddons = yield this.getAddons(oldApp);
-    let pgdbs = allAddons[0];
-    let addons = allAddons[1];
-    if (addons.length > 0) {
-      for (let addon of addons) {
-        yield this.addAddon(newApp, addon);
-      }
-    }
-    if (!skipPG && pgdbs.length > 0) {
-      let fromConfig = yield this.heroku.apps(oldApp.name).configVars().info();
-      for (let db of pgdbs) {
-        let toDB = yield this.addAddon(newApp, db);
+    let addons = yield this.heroku.get(`/apps/${oldApp.name}/addons`);
+    let attachments = yield this.heroku.get(`/apps/${oldApp.name}/addon-attachments`);
+    for (let attachment of attachments) {
+      /*jshint -W083 */
+      let addon = addons.find(a => a.id === attachment.addon.id);
+      if (addon.plan.name.startsWith('heroku-postgresql')) {
+        if (skipPG) continue;
+        let fromConfig = yield this.heroku.apps(oldApp.name).configVars().info();
+        let toDB = yield this.addAddon(newApp, addon, attachment);
         if (toDB) {
           let toConfig = yield this.heroku.apps(newApp.name).configVars().info();
-          yield this.postgres.migrateDB(fromConfig, db, toConfig, toDB);
+          yield this.postgres.migrateDB(fromConfig, addon, toConfig, toDB);
         }
+      } else if (addon.app.name !== oldApp.name) {
+        yield this.attachAddon(newApp, addon, attachment);
+      } else {
+        yield this.addAddon(newApp, addon, attachment);
```

Comment:
could you setup an app to fork for me that has this two instance issue? I tried with my forker app but it didn't seem to happen:

```
~/s/g/h/heroku-fork:(attachments) heroku addons -a forker

Add-on                                           Plan       Price
───────────────────────────────────────────────  ─────────  ──────────────────────────────
codeship (codeship-symmetrical-54012)            free       free
 └─ as CODESHIP

heroku-postgresql (postgresql-rectangular-2230)  hobby-dev  free
 └─ as DATABASE

heroku-postgresql (postgresql-regular-58206)     hobby-dev  free
 └─ as fooobar

heroku-redis (redis-defined-2951)                hobby-dev  free
 ├─ as HEROKU_REDIS_JADE
 └─ as REDIS on dickeyxxx-test app

heroku-redis (redis-rigid-2920)                  hobby-dev  free
 └─ as REDIS

heroku-redis (redis-tapered-57897)               hobby-dev  (billed to dickeyxxx-test app)
 ├─ as HEROKU_REDIS_COPPER
 ├─ as HEROKU_REDIS_WHITE
 ├─ as SECONDREDIS
 ├─ as DICKEYREDIS on dickeyxxx-test app
 ├─ as HEROKU_REDIS_COPPER on forkee app
 ├─ as HEROKU_REDIS_WHITE on forkee app
 └─ as SECONDREDIS on forkee app

The table above shows add-ons and the attachments to the current app (forker) or other apps.
```

```
~/s/g/h/heroku-fork:(attachments) heroku apps:destroy -a forkee -c forkee; heroku fork --from forker --to forkee
Destroying forkee (including all add-ons)... done
Forking forker... done. Forked to forkee
Setting buildpacks... done
Deploying 426a48a to forkee... done
Adding addon codeship:free to forkee as CODESHIP... done
Adding addon heroku-postgresql:hobby-dev to forkee as DATABASE... done
Transferring forker:DATABASE to forkee:DATABASE...
Progress: done
Adding addon heroku-postgresql:hobby-dev to forkee as fooobar... done
Transferring forker:FOOOBAR to forkee:FOOOBAR...
Progress: done
Attaching addon HEROKU_REDIS_COPPER from dickeyxxx-test to forkee as HEROKU_REDIS_COPPER... done
Adding addon heroku-redis:hobby-dev to forkee as HEROKU_REDIS_JADE... done
Attaching addon HEROKU_REDIS_WHITE from dickeyxxx-test to forkee as HEROKU_REDIS_WHITE... done
Adding addon heroku-redis:hobby-dev to forkee as REDIS... done
Attaching addon SECONDREDIS from dickeyxxx-test to forkee as SECONDREDIS... done
Copying config vars:
  LANG
  RAILS_ENV
  RACK_ENV
  SECRET_KEY_BASE
  RAILS_SERVE_STATIC_FILES
  plus
  slash
  SLASH
  JSON
  ... done
Fork complete. View it at https://forkee.herokuapp.com/
```

```
~/s/g/h/heroku-fork:(attachments) heroku addons -a forkee

Add-on                                           Plan       Price
───────────────────────────────────────────────  ─────────  ──────────────────────────────
codeship (codeship-parallel-58795)               free       free
 └─ as CODESHIP

heroku-postgresql (postgresql-closed-19138)      hobby-dev  free
 └─ as fooobar

heroku-postgresql (postgresql-concentric-60633)  hobby-dev  free
 └─ as DATABASE

heroku-redis (redis-elliptical-31005)            hobby-dev  free
 └─ as REDIS

heroku-redis (redis-transparent-37463)           hobby-dev  free
 └─ as HEROKU_REDIS_JADE

heroku-redis (redis-tapered-57897)               hobby-dev  (billed to dickeyxxx-test app)
 ├─ as HEROKU_REDIS_COPPER
 ├─ as HEROKU_REDIS_WHITE
 ├─ as SECONDREDIS
 ├─ as DICKEYREDIS on dickeyxxx-test app
 ├─ as HEROKU_REDIS_COPPER on forker app
 ├─ as HEROKU_REDIS_WHITE on forker app
 └─ as SECONDREDIS on forker app

The table above shows add-ons and the attachments to the current app (forkee) or other apps.

```